MaterialDesignThemes.Wpf/Stepper.cs

```diff
@@ -0,0 +1,345 @@
+            CommandBindings.Add(new CommandBinding(CancelCommand, CancelHandler, CanExecuteCancel));
+            CommandBindings.Add(new CommandBinding(ContinueCommand, ContinueHandler, CanExecuteContinue));
+            CommandBindings.Add(new CommandBinding(StepSelectedCommand, StepSelectedHandler, CanExecuteStepSelectedHandler));
+        }
+
+        private void LoadedHandler(object sender, RoutedEventArgs args)
+        {
+            _controller.PropertyChanged += PropertyChangedHandler;
+
+            // there is no event raised if the Content of a ContentControl changes
+            //     therefore trigger the animation in code
+            PlayHorizontalContentAnimation();
+        }
+
+        private void UnloadedHandler(object sender, RoutedEventArgs args)
+        {
+            _controller.PropertyChanged -= PropertyChangedHandler;
+        }
+
+        private bool ValidateActiveStep()
+        {
+            IStep step = _controller.ActiveStepViewModel?.Step;
+
+            if (step != null)
+            {
+                // call the validation method on the step itself
+                step.Validate();
+
+                // raise the event and call the command
+                StepValidationEventArgs eventArgs = new StepValidationEventArgs(StepValidationEvent, this, step);
+                RaiseEvent(eventArgs);
+
+                if (StepValidationCommand != null && StepValidationCommand.CanExecute(step))
+                {
+                    StepValidationCommand.Execute(step);
+                }
+
+                // the event handlers can set the validation state on the step
+                return !step.HasValidationErrors;
+            } else
+            {
+                // no active step to validate
+                return true;
+            }
+        }
+
+        private static void StepsChangedHandler(DependencyObject obj, DependencyPropertyChangedEventArgs args)
+        {
+            Stepper stepper = (Stepper)obj;
+            stepper.Controller.InitSteps(args.NewValue as IList<IStep>);
+        }
+
+        private void CanExecuteBack(object sender, CanExecuteRoutedEventArgs args)
+        {
+            args.CanExecute = true;
+        }
+
+        private void BackHandler(object sender, ExecutedRoutedEventArgs args)
+        {
+            bool isValid = ValidateActiveStep();
+
+            if (BlockNavigationOnValidationErrors && !isValid)
+            {
+                return;
+            }
+
+            _controller.Back();
+
+            args.Handled = true;
+        }
+
+        private void CanExecuteCancel(object sender, CanExecuteRoutedEventArgs args)
+        {
+            args.CanExecute = true;
+        }
+
+        private void CancelHandler(object sender, ExecutedRoutedEventArgs args)
+        {
+            if (args.Handled)
+            {
+                return;
+            }
+
+            args.Handled = true;
```

Comment:
I used the CommandManager to listen to the CancelCommand, but the event never reached my code and i think it is because of this line. If the args are set handled = true, it will no longer bubble up the tree.
In case that i am using it wrong, feel free to explain how it should be use and maybe add it to the StepperDemo.